edx_notifications/stores/sql/store_provider.py

```diff
@@ -184,6 +186,9 @@ def _get_prepaged_notifications(self, user_id, filters=None, options=None):
         if type_name:
             query = query.filter(msg__msg_type=type_name)
 
+        if start_date and end_date:
+            query = query.filter(created__gte=start_date, created__lt=end_date + timedelta(days=1))
```

Comment:
we should drop the timedelta. start_date and end_date should specify the exact range to query against.